### PR TITLE
feat(sdk): add deprecation warning for AiPlatformClient

### PR DESCRIPTION
**Pull request by ji-yaqi**
AiPlatformClient will be deprecated in SDK release 1.9 (target end of September). 

Current AiPlatformClient are recommended to switch over to use [Vertex SDK's](https://github.com/googleapis/python-aiplatform) `PipelineJob` through
```
pip install google-cloud-aiplatform
```

### CODE REVIEW

File: sdk/python/kfp/v2/google/client/client.py [hunk 1]
```diff
@@ -167,6 +167,10 @@ def __init__(
       endpoint: AIPlatformPipelines service endpoint. Defaults to
         'us-central1-aiplatform.googleapis.com'.
     """
+    logging.warning(
```
Comment (chensun) [inline sdk/python/kfp/v2/google/client/client.py:170]: nit: use `FutureWarning`, which is not ignored by default. https://docs.python.org/3/library/warnings.html#warning-categories

e.g.:
https://github.com/kubeflow/pipelines/blob/78067d1dd72d14814146851eb476667ece5f88a9/sdk/python/kfp/dsl/_container_op.py#L1176

File: sdk/python/kfp/v2/google/client/client.py [hunk 2]
```diff
@@ -167,6 +167,10 @@ def __init__(
...
+        'AIPlatformClient will be deprecated in v1.9. Please use PipelineJob in '
+        'Vertex SDK. Install the SDK using "pip install google-cloud-aiplatform"')
```
[the remark this answers is not in the document]
Comment (ji-yaqi) [reply sdk/python/kfp/v2/google/client/client.py]: Currently the notebooks/samples are not ready. I added a link pointing to the code and will update this text once the docs are ready.

File: sdk/python/kfp/v2/google/client/client.py (hunk 1 above)
Comment (ji-yaqi) [reply sdk/python/kfp/v2/google/client/client.py]: Done :)

File: sdk/python/kfp/v2/google/client/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chensun) [reply sdk/python/kfp/v2/google/client/client.py]: SG. Thanks!

**Pull request merged**